### PR TITLE
[1586] Improvements to applications open from

### PR DESCRIPTION
**Pull request by fofr**
### Context

A draft PR indicating potential improvements to Course#applications_open_from based on @benilovj's branch.

"Applications open from" should be the earliest date in any of the sites. Currently it's restricted to looking at just the sites with open applications.

This means that no date is available if applications haven't opened yet.
The wrong date might show if some of the sites have no vacancies.
No date available if the site is new and not yet running.

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by fofr** [comment]
[Applications open from shouldn't be restricted to just sites that are open to applications](https://trello.com/c/l0t1Oun3/1699-applications-open-from-shouldnt-be-restricted-to-just-sites-that-are-open-to-applications)

File: app/serializers/search_and_compare/course_serializer.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ class CourseSerializer < ActiveModel::Serializer
     # Campuses_related_Mapping
     attribute(:Campuses)                              { campuses }
     # using server date time not utc, so it's local date time?
-    attribute(:ApplicationsAcceptedFrom)              { object.applications_open_from.to_date.strftime('%Y-%m-%dT%H:%M:%S') }
+    attribute(:ApplicationsAcceptedFrom)              { object.applications_open_from.strftime('%Y-%m-%dT%H:%M:%S') }
```
Comment (defong) [inline app/serializers/search_and_compare/course_serializer.rb]:
```
def campuses
   object.site_statuses.findable.map do |site_status|
```

I guess that needs to change as well?

File: app/models/course.rb [hunk 2]
```diff
@@ -149,13 +149,17 @@ def open_for_applications?
 
   def applications_open_from
     site_statuses
-      .open_for_applications
+      .findable_or_new
+      .with_vacancies
```
Comment (defong) [inline app/models/course.rb]: > This means that no date is available if applications haven't opened yet.
The wrong date might show if some of the sites have no vacancies.
No date available if the site is new and not yet running.

Not reading last time or this time properly....

But I guess 

```
Get me the applications open from date from the course
where at least one site statuses is both
findable or new
with vacancies

and grab the earliest
```

hmmm 
why find or `new`?

Given that there `new` is just a transitional state that gets flipped when this is triggered?

https://github.com/DFE-Digital/manage-courses-backend/blob/6fb4326919b4ba1f1c51af633abcf87ee0ac8297/app/controllers/api/v2/courses_controller.rb#L37

**Comment by misaka** [comment]
@dankmitchell Can we actually remove the sidebar stuff from this PR and do it with a separate ticket? Just because of the logic behind published/not published, etc. We're starting to ticket up the provider enrichment stuff now anyway.

**Comment by NyWeb** [comment]
Closing this as it's outdated. We'll have to do the underlying applications_open_from eventually.

**Pull request closed without merging**